### PR TITLE
New configurations for Derecho 

### PR DESCRIPTION
**Pull request by mark-a-potts**
### Summary

Added new configuration files for derecho (new NCAR machine)

### Testing

The unified-dev environment builds successfully with intel and intel-mpi. No tests have yet been run against the stack, however.

### Applications affected

All

### Systems affected

New system being added

### Dependencies

None.

### Checklist
- [X] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: configs/sites/derecho/packages.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      prefix: /usr
+  tar:
+    externals:
+    - spec: tar@1.34
+      prefix: /usr
+  boost:
+    buildable: False
+    externals:
+    - spec: boost@1.78.0
+      prefix: /glade/u/home/mpotts/boost
```
Comment (climbfuji) [inline configs/sites/derecho/packages.yaml]: We can't have user directories in the site configs. Why do we need a separate boost and doxygen?

Comment (mark-a-potts) [reply configs/sites/derecho/packages.yaml]: They wouldn't build with spack for me, so I built/downloaded them directly. We don't have access to the role account yet, so I had to do it under my user name for the time being.

Comment (mark-a-potts) [reply configs/sites/derecho/packages.yaml]: And, wouldn't you know it. Just after I typed that, an email came in with access to the role account!

Comment (climbfuji) [reply configs/sites/derecho/packages.yaml]: Let's see if we can fix the boost issues somehow. doxygen might be available from the OS?

**Comment by climbfuji** [comment]
This PR needs to go to develop, but only after the release branch 1.4.1 has been merged back (#671). I changed the base branch already.

**Comment by climbfuji** [comment]
@mark-a-potts Please merge develop into your branch, thanks!

**Comment by climbfuji** [comment]
I am ok with merging this as-is (as long as we don't list it in the documentation yet as fully supported tier-1 platform, for this I would like to have boost ecflow etc built correctly). But as long as it serves your purpose for testing, it's fine.

**Review by climbfuji — APPROVED** [review]

**Comment by climbfuji** [comment]
@ulmononian Please review, and if satisifed merge it (I will be on vacation).

**Review by ulmononian — APPROVED** [review]
i've heard sunsets in cheyenne are beautiful 🤠🛤️🌄

**Pull request merged**